src/expr/skolem_manager.h

```diff
@@ -97,21 +103,30 @@ class ProofSkolemCache
    * If the variable v is not in the bound variable list of q, then null is
    * returned and an assertion failure is thrown.
    */
-  static Node mkSkolemExists(Node v,
-                             Node q,
-                             const std::string& prefix,
-                             const std::string& comment = "",
-                             int flags = NodeManager::SKOLEM_DEFAULT);
+  Node mkSkolemExists(Node v,
+                      Node q,
+                      const std::string& prefix,
+                      const std::string& comment = "",
+                      int flags = NodeManager::SKOLEM_DEFAULT,
+                      ProofGenerator* pg = nullptr);
   /**
    * Same as above, but for special case of (witness ((x T)) (= x t))
    * where T is the type of t. This skolem is unique for each t, which we
    * implement via an attribute on t. This attribute is used to ensure to
    * associate a unique skolem for each t.
+   *
+   * Notice that a purification skolem is trivial to justify, and hence it
+   * does not require a proof generator.
```

Comment:
Is this worth the special casing?

Reply:
What do you mean?

Reply:
My impression is that when using the skolem manager I'd have to check whether I have a proof generator or I have to compute myself (in this straightforward case). Wouldn't it be better to be uniform? This should not be saving much.

Reply:
Then this class would be harder to use.  I would have to update everyone who wants to construct a purification variable to construct a dummy proof generator that does the trivial justification.

Reply:
I see. In that case the other special case is better :)